test/unittests/evm_memory_test.cpp

```diff
@@ -238,3 +240,47 @@ TEST_P(evm, memory_access)
         }
     }
 }
+
+TEST_P(evm, mcopy)
+{
+    rev = EVMC_CANCUN;
+    bytecode s;
+    s += mstore(0, push(0x0123456789abcdef000000000000000000000000000000000000000000000000_u256)) +
+         mcopy(32, 0, 8) + ret(32, 8);
+    execute(s);
+    EXPECT_EQ(result.status_code, EVMC_SUCCESS);
+    ASSERT_EQ(result.output_size, 8);
+    EXPECT_EQ(bytes_view(&result.output_data[0], 8), "0123456789abcdef"_hex);
+
+    // copy uninitialized memory
```

Comment:
Add a test where all copy is from uninitialized memory.

Reply:
added